src/wallet/wallet.cpp

```diff
@@ -6357,31 +6436,23 @@ std::optional<libzcash::UnifiedAddress> UnifiedAddressForReceiver::operator()(co
         auto ufvk = wallet.GetUnifiedFullViewingKey(ufvkid);
         assert(ufvk.has_value() && ufvk.value().GetSaplingKey().has_value());
 
-        diversifier_index_t j;
         // If the wallet is missing metadata at this UFVK id, it is probably
         // corrupt and the node should shut down.
         const auto& metadata = wallet.mapUfvkAddressMetadata.at(ufvkid);
-        librustzcash_sapling_diversifier_index(
-                ufvk.value().GetSaplingKey().value().dk.begin(),
-                saplingAddr.d.begin(),
-                j.begin());
-        auto receivers = metadata.GetReceivers(j);
-        if (receivers.has_value()) {
-            auto addr = ufvk.value().Address(j, receivers.value());
-            auto addrPtr = std::get_if<std::pair<UnifiedAddress, diversifier_index_t>>(&addr);
-            if (addrPtr == nullptr) {
-                return std::nullopt;
-            } else {
-                return addrPtr->first;
+        auto saplingKey = ufvk.value().GetSaplingKey();
+        if (saplingKey.has_value()) {
```

Comment:
This conditional is doing nothing, because we assert above that `ufvk.value().GetSaplingKey().has_value()`. Either remove that assertion, or remove this conditional.

Reply:
Good point; I think I'll remove the assertion because that's more future-proof.